django/db/models/query.py

```diff
@@ -386,8 +386,15 @@ def aggregate(self, *args, **kwargs):
         query = self.query.chain()
         for (alias, aggregate_expr) in kwargs.items():
             query.add_annotation(aggregate_expr, alias, is_summary=True)
-            if not query.annotations[alias].contains_aggregate:
+            annotation = query.annotations[alias]
+            if not annotation.contains_aggregate:
                 raise TypeError("%s is not an aggregate expression" % alias)
+            for expr in annotation.get_source_expressions():
+                if expr.contains_aggregate and hasattr(expr, 'refs') and expr.refs in kwargs:
+                    name = expr.refs
+                    raise exceptions.FieldError("Cannot compute %s('%s'): '%s' is an aggregate" % (
+                        annotation.name, name, name))
+
```

Comment:
Use hanging indentation:
```suggestion
                    raise exceptions.FieldError(
                        "Cannot compute %s('%s'): '%s' is an aggregate"
                        % (annotation.name, name, name)
                    )
```